aptos-move/framework/aptos-framework/doc/object.md

```diff
@@ -0,0 +1,1239 @@
+</dt>
+<dd>
+
+</dd>
+<dt>
+<code>transfer_events: <a href="event.md#0x1_event_EventHandle">event::EventHandle</a>&lt;<a href="object.md#0x1_object_TransferEvent">object::TransferEvent</a>&gt;</code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_ObjectGroup"></a>
+
+## Struct `ObjectGroup`
+
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_ObjectGroup">ObjectGroup</a>
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>dummy_field: bool</code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_ObjectId"></a>
+
+## Struct `ObjectId`
+
+Type safe way of designate an object as at this address
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_ObjectId">ObjectId</a> <b>has</b> <b>copy</b>, drop, store
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>inner: <b>address</b></code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_CreatorRef"></a>
+
+## Struct `CreatorRef`
+
+This is a one time ability given to the creator to configure the object as necessary
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_CreatorRef">CreatorRef</a> <b>has</b> drop
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>self: <a href="object.md#0x1_object_ObjectId">object::ObjectId</a></code>
+</dt>
+<dd>
+
+</dd>
+<dt>
+<code>can_delete: bool</code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_DeleteRef"></a>
+
+## Struct `DeleteRef`
+
+The owner of this ability can delete an object
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_DeleteRef">DeleteRef</a> <b>has</b> drop, store
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>self: <a href="object.md#0x1_object_ObjectId">object::ObjectId</a></code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_ExtendRef"></a>
+
+## Struct `ExtendRef`
+
+Access to this ref allows for creation of the signer to add new resources
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_ExtendRef">ExtendRef</a> <b>has</b> drop, store
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>self: <a href="object.md#0x1_object_ObjectId">object::ObjectId</a></code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_TransferRef"></a>
+
+## Struct `TransferRef`
+
+Access to this ref allows for owner to transfer the resources between two addresses
+
+
+<pre><code><b>struct</b> <a href="object.md#0x1_object_TransferRef">TransferRef</a> <b>has</b> drop, store
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>self: <a href="object.md#0x1_object_ObjectId">object::ObjectId</a></code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="0x1_object_LinearTransferRef"></a>
+
+## Struct `LinearTransferRef`
+
+Access to this ref allows for owner to transfer the resources between two addresses
```

Comment:
should this say `transfer between two addresses once`? Why does this exist at this level, vs being left up to those that decide to use objects?